hoomd/logging.py

```diff
@@ -464,7 +465,11 @@ def from_tuple(cls, entry):
         return cls(entry[0], method, category)
 
     def __call__(self):
-        attr = getattr(self.obj, self.attr)
+        try:
+            attr = getattr(self.obj, self.attr)
+        except AttachedDataError:
+            attr = None
```

Comment:
Why necessary to assign ``None`` to ``attr`` as opposed to leaving unassigned, as in the other cases?

Reply:
Because a number of facilities, like `write_debug_data` are designed assuming that a Logger can be evaluated even if the simulation is not attached. The previous behavior was to log `None` in this case, so this maintains that.